openedx/features/content_type_gating/field_override.py

```diff
@@ -35,6 +35,21 @@ def get(self, block, name, default):
         with disable_overrides():
             original_group_access = block.group_access
 
+        # The line directly above this will set the group access field to "dirty" with
+        # the value read from the course content. Since most content does not have any
+        # value for this field it will usually be the default empty dict. This field
+        # override changes the value, however, resulting in the LMS thinking that the
+        # field data needs to be written back out to the store. This doesn't work,
+        # however, since this is a read-only setting in the LMS context. After this
+        # function returns, the _dirty_fields dict will be set correctly to contain
+        # the value from this field override. This prevents the system from attempting
+        # to save the overridden value when it thinks it has changed from the default.
+        try:
+            group_access_field = block.fields['group_access']
+            del block._dirty_fields[group_access_field]  # pylint: disable=protected-access
+        except KeyError:
+            pass
```

Comment:
I tried to come up with a better way to read the field value without setting the dirty bit, but honestly, I think this is cleaner than any of the ways I could come up with. You could avoid the try-catch with `block._dirty_fields.pop(group_access_field)`, though.